tripy/examples/resnet50/resnet50.ipynb

```diff
@@ -0,0 +1,483 @@
+    "    for k, v in pretrained_state_dict.items():\n",
+    "        if \"num_batches_tracked\" in k:\n",
+    "            continue  # Skip num_batches_tracked since it is not needed for\n",
+    "            # inference (tp.BatchNorm does not support)\n",
+    "\n",
+    "        assert k in model_state_dict_keys\n",
+    "        model_dict[k] = tp.Parameter(v.contiguous())\n",
+    "\n",
+    "    model.load_state_dict(model_dict)\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "# Load pretrained weights to tripy model\n",
+    "tripy_model = load_pretrained_weights(\n",
+    "    tripy_model,\n",
+    "    pretrained_state_dict=resnet_pretrained.state_dict(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compile Tripy Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def resnet_classifier(input):\n",
+    "    \"\"\"Inference function to compile. Runs forward path\n",
+    "    and returns argmax of output logits.\"\"\"\n",
+    "    logits = tripy_model(input)\n",
+    "    return tp.argmax(logits)\n",
+    "\n",
+    "\n",
+    "input_shape = [1, 3, 224, 224]\n",
+    "compiled_resnet_classifier = tp.compile(\n",
+    "    resnet_classifier,\n",
+    "    args=[tp.InputInfo(input_shape, dtype=tp.float32)],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tripy Demo\n",
+    "\n",
+    "Now we will test our ResNet50 classification model on a few sample images for image classification task."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import AutoImageProcessor\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "\n",
+    "# Sample image indices to classify\n",
+    "idxs = [6, 44, 34, 25, 105]\n",
+    "labels = [\n",
+    "    \"apron\",\n",
+    "    \"medicine chest, medicine cabinet\",\n",
+    "    \"mountain bike, all-terrain bike, off-roader\",\n",
+    "    \"lifeboat\",\n",
+    "    \"disk brake, disc brake\",\n",
+    "]\n",
+    "\n",
+    "# Load the image dataset\n",
+    "dataset = load_dataset(\"comet-team/coco-500\", split=\"train\")\n",
+    "\n",
+    "# Load the image processor for image pre-processing\n",
+    "processor = AutoImageProcessor.from_pretrained(\"microsoft/resnet-50\")\n",
+    "\n",
+    "fig, axes = plt.subplots(1, len(idxs), figsize=(15, 3))\n",
+    "\n",
+    "# Loop through each index and corresponding subplot\n",
+    "for ax, nid, label in zip(axes, idxs, labels):\n",
+    "    image = dataset[nid][\"Image\"]\n",
+    "    ax.imshow(image)\n",
+    "\n",
+    "    # Pre-process the image for inference\n",
+    "    processed_image = processor(image, return_tensors=\"np\")[\"pixel_values\"]\n",
```

Comment:
Can we return torch GPU tensors here? Then we can avoid a host->device copy.

Reply:
Do you mean "if processor can return GPU tensors" or moving it after processor returns?

Reply:
I am not sure if that's possible, because `processor` is not a torch model https://github.com/huggingface/transformers/blob/052e652d6d53c2b26ffde87e039b723949a53493/src/transformers/models/convnext/image_processing_convnext.py#L51 and the `preprocess` function does not take any device related args

Reply:
I'm assuming it would still respect [set_default_device](https://pytorch.org/docs/stable/generated/torch.set_default_device.html#torch-set-default-device) though? IIUC it changes the global default for all torch calls

Reply:
Ok yes, but they do a conversion to `np.array` [here](https://github.com/huggingface/transformers/blob/052e652d6d53c2b26ffde87e039b723949a53493/src/transformers/models/convnext/image_processing_convnext.py#L283) and then back to torch tensor using `torch.from_numpy` which does not respect `set_default_device`. I validated it like below

<img width="497" alt="image" src="https://github.com/user-attachments/assets/51629140-e481-4548-86f2-4426075985d4">

Reply:
Ah ok, I guess we can't avoid the copy then